### PR TITLE
Feature/gal 380

### PR DESCRIPTION
**Pull request by chenglawson1**
## Description
Lambda to load or amend user accounts.

Lambda triggered by csv file upload to bucket <env>-user-accounts-bucket. Lambda will add or amend user accounts in the <env>-UserAccounts table depending on whether the primary key UUID is new or found.

## Context

This is to allow user accounts to be added or amended.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by andrew-cleveland — APPROVED** [review]

**Review by andrew-cleveland — DISMISSED** [review]

File: terraform/src/userAccountsLambda/package.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "userAccountsLambda",
+  "version": "1.0.0",
+  "type": "module",
+  "main": "userAccountsLambda.js",
+  "author": "",
+  "dependencies": {
+    "csv-parser": "^3.0.0"
```
Comment (zainm97) [inline terraform/src/userAccountsLambda/package.json:8]: noticed that aws-sdk/client-s3 and aws-sdk/client-dynamodb are used in the lambda not mentioned in the package.json. Are the imports passing?

File: terraform/src/userAccountsLambda/userAccountsLambda.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+}
+from "@aws-sdk/client-dynamodb";
+import {
+  Readable
+}
+from 'stream';
+import csv from 'csv-parser';
+
+const s3 = new S3Client();
+const dbClient = new DynamoDBClient({ convertEmptyValues: true });
```
Comment (zainm97) [inline terraform/src/userAccountsLambda/userAccountsLambda.js:18]: this is really neat!

File: terraform/src/userAccountsLambda/userAccountsLambda.js
Comment (zainm97) [inline terraform/src/userAccountsLambda/userAccountsLambda.js:1]: just a comment on the methods. We usually have the handler at the top, after the imports and any const declarations and the methods below that

File: terraform/src/userAccountsLambda/package.json (hunk 1 above)
Comment (chenglawson1) [reply terraform/src/userAccountsLambda/package.json:8]: Lambda is working in dev. aws-sdk is loaded by AWS by default so not required in package.json.

File: terraform/src/userAccountsLambda/userAccountsLambda.js (hunk 2 above)
Comment (chenglawson1) [reply terraform/src/userAccountsLambda/userAccountsLambda.js:18]: Thanks, will resolve comment.

File: terraform/src/userAccountsLambda/package.json (hunk 1 above)
Comment (zainm97) [reply terraform/src/userAccountsLambda/package.json:8]: Cool, perfect

**Review by zainm97 — APPROVED** [review]

**Review by Abdul-NHS — APPROVED** [review]

**Pull request merged**